apps/web/components/settings/platform/hooks/usePlatformMe.ts

```diff
@@ -6,14 +6,26 @@ export const usePlatformMe = () => {
   const QUERY_KEY = "get-platform-me";
   const platformMeQuery = useQuery<UserResponse>({
     queryKey: [QUERY_KEY],
-    queryFn: async (): Promise<UserResponse> => {
+    retryOnMount: false,
+    staleTime: 300000,
+    retry: (failureCount, error) => {
+      if (error.message === "Internal Server Error") {
+        return false;
+      }
+      return failureCount < 3; // Retry up to 3 times for other errors
+    },
+    queryFn: async () => {
       const response = await fetch(`/api/v2/me`, {
         method: "get",
         headers: { "Content-type": "application/json" },
       });
-      const data = await response.json();
 
-      return data.data as UserResponse;
+      if (response.status === 200) {
+        const data = await response.json();
+        return data.data as UserResponse;
```

Comment:
Might be a good idea to use zod instead.

Reply:
zod is very heavy for client bundle size. I wouldn't introduce it only for this